db/migrate/20120219161649_add_user_image_fingerprint.rb

```diff
@@ -1,4 +1,7 @@
 class AddUserImageFingerprint < ActiveRecord::Migration[4.2]
+  class User < ActiveRecord::Base
+  end
+
```

Comment:
I don't think this will work if there are any users in the database because the migration relies on being able to access the images attached to the user via the paperclip attachment, which has not been declared here.

That of course gets even more fun once we remove paperclip from the Gemfile...

Reply:
So what should we do? Breakages from removing paperclip in the near future is the kind of fragility that comes with relying on current model code in migrations.

Of course one option is to reimplement the migration so that it doesn't depend on paperclip at all, i.e. writing some custom code to handle the fingerprint generation. This allows us to remove paperclip yet keeping this migration 100% functional. But it hardly seems worthwhile us of time. 

The second option would be to assume (like I do) that nobody is running that old a version of the database so it's all hypothetical anyway and keep what is written here. It'll work for an empty database but not one with data in it. Same situation will happen when we remove paperclip, even without this change.

And the third option, if we think migrations from this age of database are all hypothetical anyway, is to just ditch these old migrations completely.

Reply:
Well I guess we could merge this for now.

Long term I'm not sure I have any better ideas than your one of rolling up all the old migrations, though that is a lot of work to get right... The final result would have to reuse an existing timestamp to make sure existing users didn't try and apply it.